snuba/query/snql/__init__.py

```diff
@@ -82,26 +97,72 @@
 )
 
 
+class AndTuple(NamedTuple):
```

Comment:
Is there a reason you need separate classes for And and Or?

Reply:
Yes - in visit_and_expression and visit_or_expression, having these classes is really useful because it allows me to break down the logic in those visitor functions into 3 cases. With this, there's no ambiguity and the correct Expression can be built in all 3 cases.